twoliter/embedded/rpm2kit

```diff
@@ -30,3 +35,120 @@ done
 
 createrepo_c "${KIT_DIR}"
 dnf --disablerepo '*' --repofrompath "kit,file:///${KIT_DIR}" repoquery --all
+
+WORK_DIR="$(mktemp -d)"
+TIMESTAMP="$(date +"%FT%T.%NZ")"
+FILENAME_PREFIX="${KIT:?}-${ARCH:?}-${VERSION_ID:?}-${BUILD_ID:?}"
+
+# Translate ARCH into the proper docker arch
+case "${ARCH}" in
+  "x86_64")
+  DOCKER_ARCH="amd64"
+  ;;
+  "aarch64")
+  DOCKER_ARCH="arm64"
+  ;;
+esac
+
+# Create the content layer
+mkdir -p "${WORK_DIR}/blobs/sha256"
+DIFF_ARCHIVE="${WORK_DIR}/content-layer.tar"
+tar -cf "${DIFF_ARCHIVE}" -C "${KIT_DIR}" .
+DIFF_ID="$(digest_from_file "${DIFF_ARCHIVE}")"
+gzip "${WORK_DIR}/content-layer.tar"
+CONTENT_ARCHIVE="${DIFF_ARCHIVE}.gz"
+CONTENT_DIGEST="$(digest_from_file "${CONTENT_ARCHIVE}")"
+mv "${WORK_DIR}/content-layer.tar.gz" "${WORK_DIR}/blobs/sha256/${CONTENT_DIGEST}"
+
+METADATA_JQ=$(cat <<EOF
+{
+  name: "$KIT",
+  version: "$VERSION_ID",
+  sdk: .[0].sdk,
+  kit: (
+    [ .[1] | values[] | {name: ., version: "$VERSION_ID", vendor: "$VENDOR"}]
+    + [ .[0].kit[] | {name: .name, version: .version, vendor: .vendor } ]
```

Comment:
Should kits also include the digest of their dependencies?

I suppose if they do, it would be difficult to "override" transitive dependencies during development, but it also seems like it would be surprising if the world shifted underneath you in a production setting.

This also wouldn't work for local kits.

Reply:
> Should kits also include the digest of their dependencies?

🤔 I'm not sure, but I worry about making development too painful. Ideally all the digests would be captured in `Cargo.lock` and we'd prefer to fetch by digest, so that tags could shift around until the moment they were published, and then after that any change would be detected and cause people to notice.